src/ljudge.cc

```diff
@@ -2124,12 +2133,22 @@ static j::value run_testcases(const Options& opts) {
 
   vector<j::value> results;
   results.resize(opts.cases.size());
+  bool should_skip = false;
 #ifdef _OPENMP
   #pragma omp parallel for if (opts.nthread != 1 && opts.cases.size() > 1)
 #endif
   for (int i = 0; i < (int)opts.cases.size(); ++i) {
-    j::object testcase_result = run_testcase(opts.etc_dir, opts.cache_dir, opts.user_code_path, opts.checker_code_path, opts.envs, opts.cases[i], opts.skip_checker, opts.keep_stdout, opts.keep_stderr);
-    results[i] = j::value(testcase_result);
+    if (opts.skip_on_first_failure && should_skip) {
+      j::object skipped_result;
+      skipped_result["result"] = j::value(TestcaseResult::SKIPPED);
+      results[i] = j::value(skipped_result);
+    } else {
+      j::object testcase_result = run_testcase(opts.etc_dir, opts.cache_dir, opts.user_code_path, opts.checker_code_path, opts.envs, opts.cases[i], opts.skip_checker, opts.keep_stdout, opts.keep_stderr);
+      results[i] = j::value(testcase_result);
+      if (testcase_result["result"].to_str() != TestcaseResult::ACCEPTED) {
+        should_skip = true;
```

Comment:
This seems a bit tricky for OpenMP. Although it compiles, I think it's better to make the openmp loop explicitly not mutating a variable outside the loop body.

Reply:
I see, seems better.